spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreReadOptions.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.cloud.datastore.ReadOption;
+
+import org.springframework.util.Assert;
+
+/**
+ * Options for performing reads and queries in Cloud Datastore.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+public class DatastoreReadOptions {
```

Comment:
Looks like it's just a wrapper around List<ReadOption>, not sure if it is needed.
Also, there is only one possible ReadOption: https://googlecloudplatform.github.io/google-cloud-java/google-cloud-clients/apidocs/com/google/cloud/datastore/ReadOption.EventualConsistency.html
So, not sure if we even need a list.

Reply:
This will be used to hold paging and sorting options later on (take a look at the spanner module's read and query option objects). Furthermore, all the APIs from the client lib can take varag read options, so I think we should at least support that much. In the future there may be more settings.

Reply:
`ReadOption` has a very specific meaning in Datastore around consistency. I'm not sure if it's a good idea to mix in paging and sorting into the concept.